plugins/inputs/filecount/filecount.go

```diff
@@ -244,6 +244,21 @@ func (fc *FileCount) Gather(acc telegraf.Accumulator) error {
 	return nil
 }
 
+func (fc *FileCount) resolveLink(path string) (os.FileInfo, error) {
```

Comment:
This still disregards the regular only setting, since it only calls Lstat if FollowSymlinks is false. I think it may need a bit more refactoring to get that to work as expected without aggressive repeated calls to Stat and Lstat

Reply:
Fine with me.